clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -123,6 +125,7 @@ public RecordAccumulator(int batchSize,
         String metricGrpName = "producer-metrics";
         this.free = new BufferPool(totalSize, batchSize, metrics, time, metricGrpName);
         this.incomplete = new IncompleteBatches();
+        this.splitBatches = new HashSet<>();
```

Comment:
Rather than this, would it be simpler to just have a boolean on the batch indicating whether it is off-accumulator or not?